base/Dockerfile

```diff
@@ -103,16 +95,18 @@ RUN export DEBIAN_FRONTEND=noninteractive \
     && cd .. \
     # Clean up
     && rm -rf systemd \
+    # TODO: remove pip packages as well
```

Comment:
Not sure if it's required, they are pretty small (jinja2+yaml, I think)

Reply:
I missed replying to this the first time. Yes, removing the packages is required as it's not part of what we intend to deliver by default, especially if we aren't delivering pip.

Reply:
I'm thinking that we can use multi-stage builds: one that cross-compiles all the stuff, then copy the headers/libs to the final one with pre-installed compilers without any extra deps. This should require less friction and layers.

Reply:
That would be appreciated, especially if there is caching we can do to avoid rebuilding if we already have a built version later.

Reply:
Yep, docker allows to cache multi-stage builds with buildx, that should be supported by GHA, will look into that

Reply:
I've had issues with it not properly caching before, and there is also a limit to how much can be cached, which this would fill it up really quick, so you'd probably still want to clean up as much as possible to keep the cache size needed small.

Reply:
A better way to do this would be to move the build of these artifacts to a different process that stores them elsewhere with a version number, and then the docker image pulls the versioned artifact.